src/utils/schemas/README.md

```diff
@@ -0,0 +1,13 @@
+## Generating Schemas
+
+To generate schemas, you need to use the `ajv` CLI tool with the `ajv-formats` extension.
+
+For the Dataset Registration schema, the call looks like:
+
+```
+ajv compile -s consent/src/main/resources/dataset-registration-schema_v1.json -o duos-ui/src/utils/schemas/DatasetRegistrationSchemaValidation.js --spec draft2019 -c ajv-formats --strict false --all-errors
+``` 
+
+`-s` refers to the input file. `-o` refers to the output file. `--spec` specifies the specification version. `-c` adds formats such as "date", "uri", etc. `--strict false` is needed as our schema has some unknown metadata fields. `--all-errors` is needed because, by default, the validate function only returns one error at a time.
+
+Once generated, the file needs to have the `/* eslint-disable */` flag on the first line, or it will create many linting errors.
```

Comment:
I am wondering if we should create a tiny `sh` script to do this instead.

Reply:
When should this be generated? Let's say you're running it locally - would it generate it on `npm start`? there would need to be code there for vscode to consider that function as existing - so would there be a 'placeholder' old version present that gets overridden?

Reply:
What are your thoughts on `postinstall` to initially generate and also generating on `poststart` so it keeps relatively synced? And it grabs it from the backend configured in `public/config.json`? or just always from dev? just to fully get synced on the process here

Reply:
I ended up removing it from pre-start - there are so many contexts in which `start` is ran that I felt like trying to track down all of them was difficult. In particular, integration tests were failing and I wasn't sure why. Possibly, we could target the build phase - but I feel like it might be more stable to just have documentation that schemas need to be regenerated + an easy npm script to do so.

Reply:
> One question I have - I am not exactly sure how production build processes work. Does it use `npm start` / would it trigger `pre-start`? I assume it would do a build and then just serve those files, so npm start would never be ran. This might be for the best because it won't stray from what is on git. But I am not sure.

All deployed versions are built using `Dockerfile`. There, you'll see there that we have a build phase and the copy the compiled code over to `/usr/share/nginx/html` and mount that in the base nginx image. That compiled image is pushed to GCR where it is referenced by the k8s pods with env-specific `config.json` files. So, there are two ways that we generally start the UI, locally (`npm start`) and via the compiled image. The piece that I've been concerned about is having the code in `assets/schemas` checked in via this PR since it is not tied to this repo. Ideally, that code should be generated during a build step.

Reply:
I guess I feel like there are two solutions:

1. have the code checked out in the repo & easily manually updatable (this PR)
2. update the code as part of build

I feel like updating the code as part of build is a) unstable because it relies on hitting the backend which could fail - if the process just used local files then I feel like it would be stable - and b) hard to debug if it goes wrong. Keeping it a part of the repo and easily updatable via a script ensures everyone has the same code & doesn't add any complexity to the build process that could go wrong.